APLSource/lib/_DJ_.aplo

```diff
@@ -0,0 +1,64 @@
+ _DJ_ ← {
+     ⍝ Deutsch–Jozsa algorithm.
+     ⍝ Takes a vector state in the 0 state (all qubits are in the 0 states).
+     ⍝ Applies the algorithm with the passed oracle and returns the vector state without being measured.
+     ⍝
+     ⍝ Whenever the idiom {⍵:X ⋄ Y}Bs appears, The function is checking if the ancilla qubit is 0 to not apply any SWAPs
+     ⍝
+     ⍝ ⍵[1]: Index of the ancilla qubit, if this is not passed (meaning that ⍵ is simply the vector state) the ancilla qubit will be 0
```

Comment:
Let's use ⍺ instead of passing two items through ⍵. The point is preserving as much regularity as needed to then be able to reason about program execution using linear algebra. Having a combined input in the right-hand side forces us to go harder formal territory in the future.

We should do:

⍝ ⍺: Index of the ancilla qubit, if this is not passed (meaning that ⍵ is simply the vector state) the ancilla qubit will be 0
⍝ ⍵: Initial vector state in the 0 state.
⍝ ⍺⍺: Oracle to be used in the DJ algorithm.

Reply:
I did not realize you could pass both ⍺ and ⍺⍺. Not only is it more consistent with the rest of the code, it simplified the function quite a lot